packages/ton/TonBridge.ts

```diff
@@ -0,0 +1,166 @@
+import { TonConnectError } from './exceptions/TonConnectError';
+import { TonProvider } from './TonProvider';
+import {
+  AppRequest,
+  ConnectEvent,
+  ConnectEventError,
+  ConnectItemReply,
+  ConnectRequest,
+  DeviceInfo,
+  TonConnectBridge,
+  TonConnectCallback,
+  WalletEvent,
+  WalletInfo,
+  WalletResponse,
+  WalletResponseError,
+} from './types/TonBridge';
+
+const formatConnectEventError = (error: TonConnectError): ConnectEventError => {
+  return {
+    event: 'connect_error',
+    payload: {
+      code: error.code ?? 0,
+      message: error.message,
+    },
+  };
+};
+
+/**
+ * Ton bridge implementation
+ *
+ * Based on https://docs.ton.org/develop/dapps/ton-connect/protocol & open mask
+ */
+export class TonBridge implements TonConnectBridge {
+  deviceInfo!: DeviceInfo;
+  walletInfo?: WalletInfo | undefined;
+  protocolVersion: number = 2;
```

Comment:
Do we have to keep monitoring https://docs.ton.org/develop/dapps/ton-connect/protocol to make sure this value is aligned with whatever changes they make on their end?

Reply:
Yes, if there are new changes to the protocol we can increase the number when we support those changes. This indicates max supported Ton Connect version